app/views/taxonomies/_form.html.erb

```diff
@@ -62,139 +62,38 @@
+        <%= show_resource_if_allowed(f, taxonomy, :domains) %>
+        <%= show_resource_if_allowed(f, taxonomy, :realms) %>
+      <% end %>
 
-        <% if User.current.allowed_to?(:view_domains) %>
-          <div class="tab-pane" id="domains">
-            <%= checkbox_f(f, :ignore_types, {:label => _("All domains"), :multiple => true, :onchange => 'ignore_checked(this)'}, "Domain") %>
-            <%= multiple_selects f, :domains, Domain, taxonomy.selected_or_inherited_ids[:domain_ids],
-                                 {:disabled => taxonomy.used_and_selected_or_inherited_ids[:domain_ids], :label => _('Select domains')},
-                                 {'data-mismatches' => taxonomy.need_to_be_selected_ids[:domain_ids].to_json,
-                                  'data-inheriteds' => taxonomy.inherited_ids[:domain_ids].to_json,
-                                  'data-useds' => taxonomy.used_ids[:domain_ids].to_json } %>
-          </div>
-        <% end %>
+      <%= show_resource_if_allowed(f, taxonomy, :environments) %>
+      <%= show_resource_if_allowed(f, taxonomy, :hostgroups) %>
 
-        <% if User.current.allowed_to?(:view_realms) %>
-          <div class="tab-pane" id="realms">
-            <%= checkbox_f(f, :ignore_types, {:label => _("All realms"), :multiple => true, :onchange => 'ignore_checked(this)'}, "Realm") %>
-            <%= multiple_selects f, :realms, Realm, taxonomy.selected_or_inherited_ids[:realm_ids],
-                                {:disabled => taxonomy.used_or_inherited_ids[:realm_ids], :label => _('Select realms')},
-                                {'data-mismatches' => taxonomy.need_to_be_selected_ids[:realm_ids].to_json,
-                                 'data-inheriteds' => taxonomy.inherited_ids[:realm_ids].to_json,
-                                 'data-useds' => taxonomy.used_ids[:realm_ids].to_json } %>
-          </div>
-        <% end %>
-     <% end %>
-
-     <% if User.current.allowed_to?(:view_environments) %>
-        <div class="tab-pane" id="environments">
-          <%= checkbox_f(f, :ignore_types, {:label => _("All environments"), :multiple => true, :onchange => 'ignore_checked(this)'}, "Environment") %>
-          <%= multiple_selects f, :environments, Environment, taxonomy.selected_or_inherited_ids[:environment_ids],
-                               {:disabled => taxonomy.used_and_selected_or_inherited_ids[:environment_ids], :label => _('Select environments')},
-                               {'data-mismatches' => taxonomy.need_to_be_selected_ids[:environment_ids].to_json,
-                                'data-inheriteds' => taxonomy.inherited_ids[:environment_ids].to_json,
-                                'data-useds' => taxonomy.used_ids[:environment_ids].to_json } %>
+      <% if taxonomy.is_a?(Location) && SETTINGS[:organizations_enabled] && User.current.allowed_to?(:view_organizations) %>
```

Comment:
I agree there is quite some similarity; however I think I would prefer refactoring both pages to use the same template/helper in a different PR as this isn't directly related to the original bug.